lib/timber/transports/http/hackney_client.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Timber.Transports.HTTP.HackneyClient do
+  @moduledoc """
+  An efficient HTTP client that leverages hackney, keep alive connections, and connection
+  pools to communicate with the Timber API.
+
+  ## Configuration
+
+  ```elixir
+  config :timber, :hackney_client,
+    request_options: [
+      connect_timeout: 5_000, # 5 seconds, timeout to connect
+      recv_timeout: 20_000 #  20 seconds, timeout to receive a response
+    ],
+    pool_options: [
+      timeout: 600_000, # 10 minutes, how long the connection is kept alive in the pool
+      max_connections: 10 # number of connections maintained in the pool
+    ]
+  ```
+
+  * `:request_options` - Passed to `:hackney.request(method, url, headers, body, request_options)`.
+  * `:pool_options` - Passed to `:hackney_pool.start_pool(@pool_name, pool_options)`.
+
+  """
+
+  @behaviour Timber.Transports.HTTP.Client
+
+  @pool_name __MODULE__
+  @default_request_options [
+    connect_timeout: 5_000, # 5 seconds, timeout to connect
+    recv_timeout: 20_000 #  20 seconds, timeout to receive a response
+  ]
+  @default_pool_options pool_options: [
+    timeout: 600_000, # 10 minutes, how long the connection is kept alive in the pool
+    max_connections: 10 # number of connections maintained in the pool
```

Comment:
This may be excessive. I doubt it will exceed 2 connections, and only then because of rotating a closed connection to an open one